include/ginkgo/core/base/index_set.hpp

```diff
@@ -280,7 +280,7 @@ class index_set {
      *
      * @return  number of indices stored in the index set
      */
-    index_type get_num_elems() const { return this->num_stored_indices_; };
```

Comment:
Same issue as with `device_matrix_data`.
But, TBH, I think the naming of this and `get_size` is confusing anyway.
For an index set that covers the interval `[0, n)` `get_size` returns `n`, but that doesn't need to correlate to the size of any stored arrays. So, to me, renaming it to `get_end`, or maybe just `end`, seems clearer.
As for `get_num_elems`, it's honestly really hard to tell what this is for. I think it is the sum over the length of all stored subsets. So I would assume it's the size of the array returned by `to_global_indices`, but judging from its implementation it's not??? I'm very confused...

Reply:
Perhaps `get_num_global_indices` would describe this a bit better? @pratikvn any suggestions?

Reply:
I still think this needs a better name.

Reply:
get_superset_size?

Reply:
I'll do this in a new PR